tensorflow/core/platform/env.cc

```diff
@@ -266,6 +270,13 @@ string Env::GetExecutablePath() {
   char unresolved_path[buffer_size];
   _NSGetExecutablePath(unresolved_path, &buffer_size);
   CHECK(realpath(unresolved_path, exe_path));
+#elif defined(__FreeBSD__)
+  int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+  size_t exe_path_size = PATH_MAX;
+
+  if (sysctl(mib, 4, exe_path, &exe_path_size, NULL, 0) != 0) {
+    // Not sure what to do if it fails?
```

Comment:
well, what is the nature of this failure?
What exactly are we checking for in this if block?

Reply:
If sysctl cant resolve the path it returns a non zero value, but i can't find anything about a scenario where it would fail, and I cant see any of the other OS:s checking for a failure in their resolver so I just left it unhandled, should i set exe_path to an empty value if it fails perhaps?

Or should i perhaps do a 
> return errors::Aborted("could not resolve path ", exe_path)

Reply:
Having a failure like this looks more correct to me. Let's do that.